python/langsmith/evaluation/_runner.py

```diff
@@ -870,48 +878,134 @@ def _evaluate(
     client: Optional[langsmith.Client] = None,
     blocking: bool = True,
     experiment: Optional[Union[schemas.TracerSession, str, uuid.UUID]] = None,
-) -> ExperimentResults:
+    hyper_params: Optional[Dict[str, List[Any]]] = None,
+) -> ExperimentResults | CombinedExperimentResults:
     # Initialize the experiment manager.
     client = client or rt.get_cached_client()
-    runs = None if _is_callable(target) else cast(Iterable[schemas.Run], target)
-    experiment_, runs = _resolve_experiment(
-        experiment,
-        runs,
-        client,
-    )
+    if hyper_params is None:
+        runs = None if _is_callable(target) else cast(Iterable[schemas.Run], target)
+        experiment_, runs = _resolve_experiment(
+            experiment,
+            runs,
+            client,
+        )
```

Comment:
@hinthornw not sure about this part, feel like we might need more checks and also corresponding ones inside of `aevaluate`